admin/src/multitree_bench/mod.rs

```diff
@@ -269,257 +281,443 @@ impl ChainGenerator {
 		key
 	}
 
-	fn num_node_children(&self, _tree_index: u64, depth: u32, seed: u64) -> u32 {
+	/// Returns tuple of node data and child specs. Child specs are depth and seed.
+	fn generate_node(
+		&self,
+		depth: u32,
+		seed: u64,
+		enumerated_depth_child_counts: &mut Vec<Vec<usize>>,
+	) -> (Vec<u8>, Vec<(u32, u64)>) {
 		let mut rng = rand::rngs::SmallRng::seed_from_u64(seed);
 
-		let num_children = if depth < self.depth_child_count_histograms.len() as u32 {
-			self.depth_child_count_histograms[depth as usize].sample(rng.next_u64())
-		} else {
-			0
-		};
+		let num_children =
+			enumerated_depth_child_counts.get_mut(depth as usize).map_or(None, |x| x.pop());
 
-		num_children
+		let num_children = num_children.unwrap_or_else(|| {
+			self.depth_child_count_histograms
+				.get(depth as usize)
+				.map_or(0, |h| h.sample(rng.next_u64()) as usize)
+		});
+
+		let mut children = Vec::with_capacity(num_children as usize);
+
+		for _ in 0..num_children {
+			children.push((depth + 1, rng.next_u64()));
+		}
+
+		let v = self.generate_node_data(&mut rng, num_children == 0);
+
+		(v, children)
 	}
 
-	/// Returns tuple of node data and child specs. When only_direct_children is true node data will
-	/// be empty and non direct children will be NodeSpec::UnresolvedPath.
-	fn generate_node(
+	fn generate_node_data(&self, rng: &mut rand::rngs::SmallRng, leaf: bool) -> Vec<u8> {
+		// Polkadot doesn't store actual values in branch nodes, only in leaf nodes. Hence the value
```

Comment:
Guess it is a result of the content being well addressed in polkadot, not a rule in the absolute. Maybe:
```suggestion
		// Polkadot avoid storing actual values in branch nodes, mainly in leaf nodes. Hence the value
```

Reply:
Good point. I'll change the comment.